app/controllers/Authentication.scala

```diff
@@ -118,25 +124,49 @@ class Authentication @Inject() (
     Ok(views.html.main()(Html("")))
   }
 
+  def normalizeName(name: String):Option[String] = {
+    val replacementMap = Map("ü" -> "ue", "Ü" -> "Ue", "ö" -> "oe", "Ö" -> "Oe", "ä" -> "ae", "Ä" -> "Ae", "ß" -> "ss",
+      "é" -> "e", "è" -> "e", "ê" -> "e", "È" -> "E", "É" -> "E", "Ê" -> "E",
+      "Ç" -> "C", "ç" -> "c", "ñ" -> "n", "Ñ" -> "N", "ë" -> "e", "Ë" -> "E", "ï" -> "i", "Ï" -> "I",
+      "å" -> "a", "Å" -> "A", "œ" -> "oe", "Œ" -> "Oe", "æ" -> "ae", "Æ" -> "Ae",
+      "þ" -> "th", "Þ" -> "Th", "ø" -> "oe", "Ø" -> "Oe", "í" -> "i", "ì" -> "i")
+
+    val finalName = name.map(c => replacementMap.getOrElse(c.toString,c.toString)).mkString.replaceAll("[^A-Za-z0-9_\\-\\s]", "")
+    if(finalName.isEmpty)
+      None
+    else
+      Some(finalName)
+  }
+
   def handleRegistration = Action.async { implicit request =>
-    signUpForm.bindFromRequest.fold(
-      bogusForm =>  Future.successful(BadRequest(bogusForm.toString)),
+      signUpForm.bindFromRequest.fold(
+      bogusForm => Future.successful(BadRequest(bogusForm.toString)),
       signUpData => {
         val loginInfo = LoginInfo(CredentialsProvider.ID, signUpData.email)
+        var errorMap = List[String]()
+        normalizeName(signUpData.firstName).getOrElse(errorMap ::= Messages("user.firstName.invalid"))
+        normalizeName(signUpData.lastName).getOrElse(errorMap ::= Messages("user.lastName.invalid"))
         UserService.retrieve(loginInfo).toFox.futureBox.flatMap {
           case Full(_) =>
-            Fox.successful(BadRequest(Messages("error.userExists", signUpData.email)))
+            errorMap ::= Messages("user.email.alreadyInUse")
+            Fox.successful(BadRequest(Json.obj("messages" -> Json.toJson(errorMap.map(t => Json.obj("error" -> t))))))
           case Empty =>
-            for {
-              user <- UserService.insert(signUpData.team, signUpData.email, signUpData.firstName, signUpData.lastName, signUpData.password, automaticUserActivation, roleOnRegistration,
-                                         loginInfo, passwordHasher.hash(signUpData.password))
-              brainDBResult <- BrainTracing.register(user).toFox
-            } yield {
-              Mailer ! Send(DefaultMails.registerMail(user.name, user.email, brainDBResult))
-              Mailer ! Send(DefaultMails.registerAdminNotifyerMail(user, user.email, brainDBResult))
-              Ok
-             // if (automaticUserActivation) { JsonOk(Messages("user.automaticUserActivation")) } else {JsonOk(Messages("user.accountCreated")) }
+            if(errorMap.isEmpty){
+              Fox.successful(BadRequest(Json.obj("messages" -> Json.toJson(errorMap.map(t => Json.obj("error" -> t))))))
```

Comment:
errorMap contains errors, right? why do we return the errors, when the errorMap is empty? is it possible that it should be "!errors.isEmpty", or am I getting it wrong?

Reply:
You're right it looks wrong. But I thought I tested it, will investigate tomorrow.